### DISCUSSION TITLE
Edge network cache expires before `s-maxage`

### DISCUSSION THREAD

**Discussion by mbernardes19** [vercel/community#1329] [Help]
### Summary

I have a Next v12 site deployed on Vercel, and I've created a page /sitemap.xml that uses the Next's `getServerSideProps` function to generate the sitemap dynamically and then sends the xml file as a string using the `res.write` function. I'm also setting these headers on the request before sending it:
```
Content-Type:  text/xml
Cache-Control: s-maxage=86400, stale-while-revalidate=60
```

On the first request, the response is cached correctly. Here are the headers of the first request to the page after a deployment:
```
age: 0
cache-control: public, max-age=0, must-revalidate
content-encoding: br
content-type: text/xml
date: Sun, 22 Jan 2023 22:38:31 GMT
server: Vercel
strict-transport-security: max-age=63072000
x-matched-path: /en/sitemap.xml
x-vercel-cache: MISS
x-vercel-id: gru1::iad1::mhwx7-1674427105712-cf7adaac2a6a
```
Then, on a second request right after, I'm able to confirm that it was cached as expected:

```
age: 78
cache-control: public, max-age=0, must-revalidate
content-encoding: br
content-type: text/xml
date: Sun, 22 Jan 2023 22:38:31 GMT
server: Vercel
strict-transport-security: max-age=63072000
x-matched-path: /en/sitemap.xml
x-vercel-cache: HIT
x-vercel-id: gru1::iad1::hc2gw-1674427189493-861547617697
```

But the issue is that this cache never lasts for the time specified in the page's response header `s-maxage` value (86400 seconds, 1 day). Did anyone ever face an issue like this one? Is there anything I'm missing when setting the cache time?

Please let me know if there's any more info that I should provide.

**Comment by angellromero** [comment]
I am facing this exact issue.  I have a cache rule of 48 hours and a SWR of 24 hours and I find myself fragmenting cache on the same request minutes later.  I can't see this being a user on my end as setting the cache control headers are very straightforward.  In my case, I should see a Cache type of STALE in the worst-case scenario.  

Any additional insights from Vercel would be appreciated here as its presenting some doubts on how edge caching is working on the product.
